holder.go

```diff
@@ -351,6 +359,42 @@ func (h *Holder) flushCaches() {
 	}
 }
 
+// setFileLimit attempts to set the open file limit to the FileLimit constant defined above.
+func (h *Holder) setFileLimit() {
+	lim := &syscall.Rlimit{}
+	var setLimit uint64
+	if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, lim); err != nil {
+		h.logger().Printf("ERROR checking open file limit: %s", err)
+		return
+	}
+	if lim.Max < FileLimit {
+		h.logger().Printf("WARNING: open file limit max is %d. Please consider running \"ulimit -n %d\" before starting Pilosa to avoid \"too many open files\" error.", lim.Max, FileLimit)
```

Comment:
I think we need better help messages than `ulimit -n` especially in the case below where the OS accepted the syscall, but silently refused to set the limit - the user is probably on OSX and will need better direction to fix the problem. This post https://blog.dekstroza.io/ulimit-shenanigans-on-osx-el-capitan/ describes the arduous journey to actually changing the open file limit on el capitan, I'm not sure if it works on Sierra as well, but I think we should at least try to give the user some place to look further if they are on OSX.